tests/regression/summarize_test_results.py

```diff
@@ -55,14 +55,43 @@
 }
 
 
-def get_metric_dict(dict_type, idx: int, model: str):
-    if dict_type:
-        return dict_type[idx][model]
+def get_metric_dict(dict_data: Union[List[Dict[str, Any]], None], idx: int, model: str):
+    """Get the proper dict item by referencing the index and model information.
+
+    Since all models could be optimized by POT or NNCF, we need to check that there are proper values in the data.
+    For example, if A model could be optimized by both POT and NNCF and B model couldn't be supported by POT and NNCF.
+    In this case, we have POT, NNCF results about A, however, we don't have POT, NNCF results about B.
+
+    So, if we don't have results, we need to mark the empty result as "-".
+
+    """
+    if dict_data:
```

Comment:
Current implementation also works but how about to use code as below?
```
if dict_data and len(dict_data) > idx:
    return dict_data[idx][model]
else:
    return "-"
```

Reply:
Done. [f1cc084](https://github.com/openvinotoolkit/training_extensions/pull/2095/commits/f1cc084fc482f87c3a4c0b6a11092d0ec66740d8)